internal/twirptest/service_test.go

```diff
@@ -574,6 +575,48 @@ func TestErroringHooks(t *testing.T) {
 	})
 }
 
+func TestInterceptor(t *testing.T) {
+	interceptor := func(next twirp.Method) twirp.Method {
+		return func(ctx context.Context, request interface{}) (interface{}, error) {
```

Comment:
verify that the context contains info about the request:

```go
methodName, _ := twirp.MethodName(ctx)
if methodName != "MakeHat" {
  return nil, fmt.Errorf("unexpected methodName: %q", methodName)
}
serviceName, _ := twirp.ServiceName(ctx)
if serviceName != "Haberdasher" {
  return nil, fmt.Errorf("unexpected serviceName: %q", serviceName)
}
packageName, _ := twirp.PackageName(ctx)
if packageName != "twirp.internal.twirptest" {
  return nil, fmt.Errorf("unexpected packageName: %q", packageName)
}
```

later, after calling the service, we should also verify that the context was annotated with the status code:

```go
response, err := next(ctx, request)

status, ok := twirp.StatusCode(ctx)
if status != "200" || !ok {
    return nil, fmt.Errorf("unexpected status in context: %q", status)
}
```

Reply:
Added the first part, but the second part (status on context) isn't how contexts work - contexts are immutable, and we'd have to return a new context to do this (which isn't standard). I'd expect to get the status code off of the error, not the context, and all other middleware of this variety in the Golang ecosystem does the same.

Reply:
good idea, thanks. Getting statusCode from context - doesn't work.
This is how I did:
```
twerr, ok := err.(twirp.Error)
if ok {
	statusCode = twirp.ServerHTTPStatusFromErrorCode(twerr.Code())
}
```